shuffle/cli/src/dev_api_client.rs

```diff
@@ -97,7 +102,9 @@ impl DevApiClient {
     }
 
     fn check_response_status_code(status: &StatusCode, context: &str) -> Result<()> {
-        match status >= &StatusCode::from_u16(200)? && status < &StatusCode::from_u16(300)? {
+        match status >= &StatusCode::from_u16(200)? && status < &StatusCode::from_u16(300)?
+            || status == &StatusCode::from_u16(404)?
```

Comment:
this is not right, 404 is not found, depending on context, you decide whether it is acceptable response, should not be considered as success

Reply:
Hmm. in these lines of client.ts file https://github.com/diem/diem/blob/main/shuffle/move/examples/main/client.ts#L30-L49, you're allowing the 404 error to pass. If you look at the `check_txn_executed_from_hash`, it loops just like the client.ts file and every time `get_transactions_by_hash` is called, it checks `check_response_status_code`. 


I did feel a bit uneasy making the 404 a success tho. maybe i can take care of the 404 case in another method?